### PR TITLE
Add a mutex which guards access to pods in Kubelet.

### PR DESCRIPTION
**Pull request by gmarek**
As per #4462.

### CODE REVIEW

File: pkg/kubelet/kubelet.go [hunk 1]
```diff
@@ -157,7 +157,8 @@ type Kubelet struct {
 	podInfraContainerImage string
 	podWorkers             *podWorkers
 	resyncInterval         time.Duration
-	pods                   []api.BoundPod
+	pods                   []api.BoundPod // guarded by podsMutex
+	podsMutex              sync.RWMutex
```
Comment (xiang90) [inline pkg/kubelet/kubelet.go]: move the mu before pods? and move the comment on the mu?

Comment (gmarek) [inline pkg/kubelet/kubelet.go]: I noticed that I missed that one of the function is returning pointer. I'll need to handle that as well.

**Comment by gmarek** [comment]
Added copying to guarded Get... methods and applied the fix.

File: pkg/kubelet/kubelet.go [hunk 2]
```diff
@@ -1304,6 +1305,8 @@ func (kl *Kubelet) cleanupOrphanedVolumes(pods []api.BoundPod, running []*docker
 
 // SyncPods synchronizes the configured list of pods (desired state) with the host current state.
 func (kl *Kubelet) SyncPods(pods []api.BoundPod) error {
+	kl.podsMutex.Lock()
```
Comment (wojtek-t) [inline pkg/kubelet/kubelet.go]: Is this needed? We're not accessing kl.pods in this method.

**Comment by rsokolowski** [comment]
PR #4494 addresses the same thing and is IMHO cleaner (less changes).

**Comment by rjnagal** [comment]
Closing in favor of #4494

**Pull request closed without merging**